x-pack/legacy/plugins/siem/public/utils/default_date_settings.test.ts

```diff
@@ -0,0 +1,545 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import chrome from 'ui/chrome';
+import {
+  getDefaultFromString,
+  DefaultTimeRangeSetting,
+  getDefaultToString,
+  getDefaultFromValue,
+  getDefaultToValue,
+  getDefaultFromMoment,
+  getDefaultToMoment,
+  DefaultIntervalSetting,
+  getDefaultIntervalKind,
+  getDefaultIntervalDuration,
+  parseDateString,
+  parseDateWithDefault,
+} from './default_date_settings';
+import {
+  DEFAULT_FROM,
+  DEFAULT_SIEM_TIME_RANGE,
+  DEFAULT_TO,
+  DEFAULT_SIEM_REFRESH_INTERVAL,
+  DEFAULT_INTERVAL_PAUSE,
+  DEFAULT_INTERVAL_VALUE,
+  DEFAULT_INTERVAL_TYPE,
+} from '../../common/constants';
+import { Policy } from '../store/inputs/model';
+import moment from 'moment';
+
+// Change the constants to be static values so we can test against those instead of
+// relative sliding date times. Jest cannot access these outer scoped variables so
+// we have to repeat ourselves once
+const DEFAULT_FROM_DATE = '1983-05-31T13:03:54.234Z';
+const DEFAULT_TO_DATE = '1990-05-31T13:03:54.234Z';
+jest.mock('../../common/constants', () => ({
+  DEFAULT_FROM: '1983-05-31T13:03:54.234Z',
+  DEFAULT_TO: '1990-05-31T13:03:54.234Z',
+  DEFAULT_INTERVAL_PAUSE: true,
+  DEFAULT_INTERVAL_TYPE: 'manual',
+  DEFAULT_INTERVAL_VALUE: 300000,
+  DEFAULT_SIEM_REFRESH_INTERVAL: 'siem:refreshIntervalDefaults',
+  DEFAULT_SIEM_TIME_RANGE: 'siem:timeDefaults',
+}));
+
+/**
+ * We utilize the internal chrome mocking that is built in to be able to mock different time range
+ * scenarios here or the absence of a time range setting.
+ * @param timeRange timeRange to use as a mock, including malformed data
+ */
+const mockTimeRange = (
+  timeRange: DefaultTimeRangeSetting = { from: DEFAULT_FROM, to: DEFAULT_TO },
+  interval: DefaultIntervalSetting = {
+    pause: DEFAULT_INTERVAL_PAUSE,
+    value: DEFAULT_INTERVAL_VALUE,
+  }
+) => {
+  chrome.getUiSettingsClient().get.mockImplementation((key: string) => {
+    switch (key) {
+      case DEFAULT_SIEM_TIME_RANGE:
+        return timeRange;
+      case DEFAULT_SIEM_REFRESH_INTERVAL:
+        return interval;
+      default:
+        throw new Error(`Unexpected config key: ${key}`);
+    }
+  });
+};
+
+/**
+ * Return that this unknown is only an object but we recognize in Typescript that we are ok
+ * with the object being malformed.
+ * @param timeRange Malformed object
+ */
+const isMalformedTimeRange = (timeRange: unknown): timeRange is DefaultTimeRangeSetting =>
+  typeof timeRange === 'object';
+
+/**
+ * Return that this unknown is only an object but we recognize in Typescript that we are ok
+ * with the object being malformed.
+ * @param timeRange Malformed object
```

Comment:
Parameter mismatch in function signature: 
```suggestion
 * @param interval Malformed object
```